### PR TITLE
Added a landmark viewer that shows volcanoes as a Modest Maps test monkey

### PR DESCRIPTION
**Pull request by SmilyOrg**
Fixed autowidth when drawing text boxes (previously it accidentally stored it as a text label due to both using NaN as width, label now uses -1)
Fixed string.split ignoring duplicated separators (now it adds an empty string instead as is tradition)
Fixed some MM bugs regarding changing providers (it still isn't perfect)
Added new (less broken?) haversine distance function to MM extras

### CODE REVIEW

**Comment by SmilyOrg** [comment]
Probably needs more docs and a bit more cleanup :)

File: sdk/src/loom/ModestMaps/extras/Distance.ls [hunk 1]
```diff
@@ -35,6 +37,27 @@ package loom.modestmaps.extras
                            Math.cos(a1)*Math.sin(b1)*Math.cos(a2)*Math.sin(b2) + 
                            Math.sin(a1)*Math.sin(a2)) * r;;
             return d;
-        }        
+        }
+        
+        public static function haversineDistance(start:Location, end:Location, r:Number=R_METERS):Number
```
Comment (bengarney) [inline sdk/src/loom/ModestMaps/extras/Distance.ls]: Can you document this?

File: docs/examples/ModestMapsExample/src/LandmarkViewer.ls [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+        public var region:String;
+        public var latitude:Number;
+        public var longitude:Number;
+        public var elevation:Number;
+        public var type:String;
+        public var status:String;
+        public var lastEruption:String;
+    }
+    
+    public class LandmarkViewer
```
Comment (bengarney) [inline docs/examples/ModestMapsExample/src/LandmarkViewer.ls]: Please document what this class does!

**Comment by bengarney** [comment]
Looking good. Kill dead code + add comments and it's good with me.

**Comment by bengarney** [comment]
Weird:

```
[10:21:55][Execute test] ########################################
[10:21:55][Execute test] 
[10:21:55][Execute test] ############# FAILED TESTS #############
[10:21:55][Execute test] 
[10:21:55][Execute test] ########################################
[10:21:55][Execute test] 
[10:21:55][Execute test] Failing type: tests.TestString
[10:21:55][Execute test] Tests:      1 total,    0 passed,    1 failed
[10:21:55][Execute test] Asserts:    1 total,    0 passed,    1 failed
[10:21:55][Execute test] 
[10:21:55][Execute test] ########################################
[10:21:55][Execute test] 
[10:21:55][Execute test] run   0 / 1 asserts passed
[10:21:55][Execute test] 
[10:21:55][Execute test]    1 Assert.fail  // Legacy test failed: tests.TestString
[10:21:55][Execute test]      ./src/unittest//LegacyTest.ls:fail:83
[10:21:55][Execute test]      ./src/unittest//LegacyTest.ls:end:142
[10:21:55][Execute test]      ./src/unittest//LegacyTest.ls:run:169
[10:21:55][Execute test]      ./src/unittest//TestRunner.ls:runCallback:446
[10:21:55][Execute test]      ./src/unittest//TestRunner.ls:runTypesCallback:285
[10:21:55][Execute test]      ./src/unittest//TestRunner.ls:runCallback:433
[10:21:55][Execute test]      ./src/unittest//TestRunner.ls:runHandleTest:485
[10:21:55][Execute test]      ./src/unittest//TestRunner.ls:done:635
[10:21:55][Execute test]      ./src/test//tests/AsyncTest.ls:parameter:37
[10:21:55][Execute test]      ./src/unittest//TestRunner.ls:runCallback:451
[10:21:55][Execute test]      ./src/unittest//TestRunner.ls:runTypesCallback:285
[10:21:55][Execute test] 
```

**Pull request merged**